sql/types/json_encode.go

```diff
@@ -0,0 +1,297 @@
+package types
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+	"sort"
+	"strconv"
+)
+
+type NoCopyBuilder struct {
```

Comment:
Probably could use a better name.  It's not so much that it doesn't copy.  It's that it doesn't copy every time it needs to resize the buffer like string builder does.

Reply:
Name is fine